dev/Deployment/Deployment.idl

```diff
@@ -24,14 +29,36 @@ namespace Microsoft.Windows.ApplicationModel.WindowsAppRuntime
         HRESULT ExtendedError{ get; };
     };
 
+    /// This object is used to specify deployment options to apply when using DeploymentManager's
+    /// Initialize method
+    [contract(DeploymentContract, 2)]
+    runtimeclass DeploymentInitializeOptions
```

Comment:
My understanding was that the APIs are private and the SDK handles all of this underneath it all.

Reply:
`DeploymentManager.GetStatus()` tells you if WinAppSDK's runtime is properly/fully installed and ready to rock - or not.

`DeploymentManager.Initialize()` checcks status and, if not OK, installs packages. The driving motivator behind this is Store-distributed packaged apps. Store will install framework package dependencies so WARfwk is covered...but that's it. MSIX today has no way for a package to dependency on a main package and thus Store has no way of knowing you also need WARmain+singleton. So Store-distributed packaged apps can call this (e.g. in main()) to ensure WinAppSDK's installed and ready to rock, and thus things like Push Notifications are functional :-)

It can also make for a convenient "repair" option (for any type of app), but the critical raison raison d'etre is the Store-distributed packaged app, as noted above

Reply:
> My understanding was that the APIs are private and the SDK handles all of this underneath it all.

DeploymentManager is a new WinAppSDK invention, and it only calls public APIs. Just like the installer, not special sauce anyone couldn't do for themselves - just conveniently shrinkwrapped we provide to DoTheRightThing(TM) so devs don't have to sweat the details :-)